main.py

```diff
@@ -202,6 +203,11 @@ def upload_cv_upload(person_id, signature):
 
 @app.route('/email_candidate/<int:person_id>', methods=['GET','POST'])
 def email_candidate(person_id):
+    if 'postcode' not in flask.session:
+        # jump out
+        flask.flash("Something went wrong", 'danger')
```

Comment:
Wasn’t meant to be aggressive!

> if, say, somebody tweeted the link.

What would that tweet say? “Email candidate x here: <link>”
Link then takes User 2 to an unexpected place.